packages/components/src/ClickableStyle/ClickableStyle.tsx

```diff
@@ -23,6 +23,33 @@ export type ClickableStyleProps<IComponent extends React.ElementType> = {
    * The style of the element.
    */
   variant: "flat" | "outline" | "link";
+  /**
+   * Optional icon that can be positioned to the left or right of the other button contents.
+   *
+   * Default position is 'left'. Use the iconPlacement prop to move it to the right side.
+   */
+  icon?: {
+    /*
+     * The icon component. See the full list of available icons here:
+     * https://chanzuckerberg.github.io/edu-design-system/?path=/story/svgicon--all-icons
+     */
+    iconElement: ReactNode;
+    /*
+     * Whether the icon should be on the left or right side of the button's other contents.
+     *
+     * Default is "left".
+     *
+     * If the button has no other contents, this prop will be ignored.
+     */
+    iconPlacement?: "left" | "right";
+    /*
+     * Margin to be added to icon's left and right sides.
+     *
+     * Icons typically have a little extra space between the path and the edge of the viewbox.
+     * Negative left and right margin can be used to remove this extra space.
+     */
+    iconHorizontalMargin?: any;
```

Comment:
could this be type `string`?

Reply:
Or `string | 0`

Reply:
@ahuth do you think we need to accommodate a value of `0`? It's 0 by default, right?

Reply:
Oh duh, I'm setting it to 0 🤦